File: LEETCODE/ARRAY/(Kadanes) maximum_subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        sumi=0
        maxi=nums[0]
        for i in range(len(nums)):
            sumi+=nums[i]
            if sumi > maxi:
                maxi =sumi
            if sumi<0:
                sumi=0
        return maxi
            
        """
        :type nums: List[int]
        :rtype: int
        """
        

File: LEETCODE/ARRAY/(Kadanes) maximum_subarray.java
class Solution {
    public int maxSubArray(int[] nums) { 
        if (nums.length == 1){
            return nums[0];
        }
        else{
            int sum=0;
            int maximum = nums[0];
            for (int number: nums){
                sum+=number;
                if (sum>maximum){
                    maximum=sum;
                }
                if (sum < 0){
                    sum=0;
                }
            }
            return maximum;
        }
    }
}
